src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -120,13 +121,15 @@ private void UserControl_Loaded(object sender, RoutedEventArgs e)
                         string solutionDirectory = await _solutionManager.GetSolutionDirectoryAsync(CancellationToken.None);
                         _componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
                         _vsSolutionManager = _componentModel.GetService<IVsSolutionManager>();
+                        _solutionRestoreWorker = _componentModel.GetService<ISolutionRestoreWorker>();
 
-                        // when the control is first loaded, check for missing packages
+                        // if the project is PR check for missing assets file if the user has disabled the automatic restore
```

Comment:
The comment says we're checking if automatic restores has been disabled but I don't see that in the code. Was this missed or are we checking that with the existing conditions?

Reply:
This comment in wrong, I thought that `ISolutionRestoreWorker.IsRunning` could give us enough information about the restore state, but after Nikolche [comment ](https://github.com/NuGet/NuGet.Client/pull/4549#discussion_r844442229)about the race condition I decided to change this comment and add that validation

Reply:
The comment probably has the same issue now :D

Reply:
@martinrrm 👋

Reply:
I updated the comment, it is hard to know if automatic restore has been disabled, but we can check if a restore is running and fix the issue bug